### PR TITLE
Allow custom input generators in Aggregation Fuzzer

### PR DESCRIPTION
**Pull request by mbasmanova**
Introduce InputGenerator interface to customize input generation for individual
aggregate functions. Use this to customize (1) 'n' argument for
min/max/min_by/max_by to use a constant value within [0, 10000] range, (2) 'e' 
argument for approx_distinct to use a constant value within [0.0040625, 0.26] 
range; (3) w, percentile(s) and accuracy arguments for approx_percentile to 
use valid values.

Fix approx_distinct function to allow 'e' argument to be constant, but not necessarily 
constant-encoded.

Fix approx_percentile to allow percentile(s) and accuracy arguments to be constant, 
but not necessarily constant-encoded.

Add logic to keep track of signatures that fail more than 50% of the time. This
data will help prioritize adding custom input generators.

Part of #7596

### CODE REVIEW

File: velox/exec/tests/AggregationFuzzerTest.cpp [hunk 1]
```diff
@@ -37,6 +38,111 @@ DEFINE_string(
...
+
+    // Make sure to use the same value of 'n' for all batches in a given Fuzzer
+    // iteration.
+    if (!n_.has_value()) {
+      n_ = boost::random::uniform_int_distribution<int64_t>(0, 9'999)(rng);
+    }
+
+    const auto size = fuzzer.getOptions().vectorSize;
+
+    std::vector<VectorPtr> inputs;
```
Comment (pedroerp) [inline velox/exec/tests/AggregationFuzzerTest.cpp:66]: nit: reserve to prevent reallocation

File: velox/exec/tests/AggregationFuzzerTest.cpp [hunk 2]
```diff
@@ -112,12 +218,33 @@ int main(int argc, char** argv) {
           {"sum_data_size_for_stats", ""},
       };
 
+  std::unordered_map<
+      std::string,
+      std::shared_ptr<facebook::velox::exec::test::InputGenerator>>
+      customInputGenerators;
```
Comment (pedroerp) [inline velox/exec/tests/AggregationFuzzerTest.cpp:224]: maybe use a initialization list
```
  ... customInputGenerators = {
    {"min", std::make_shared<MinMaxInputGenerator>("min")},
    ...
  }
```

File: velox/exec/tests/AggregationFuzzerTest.cpp [hunk 3]
```diff
@@ -112,12 +218,33 @@ int main(int argc, char** argv) {
...
+
+  using MinMaxInputGenerator =
```
Comment (pedroerp) [inline velox/exec/tests/AggregationFuzzerTest.cpp:226]: you can just do:

```
 using facebook::velox::exec::test::MinMaxInputGenerator;
```

File: velox/exec/tests/AggregationFuzzerTest.cpp [hunk 4]
```diff
@@ -112,12 +218,33 @@ int main(int argc, char** argv) {
...
+      std::make_shared<
+          facebook::velox::exec::test::ApproxDistinctInputGenerator>());
+
   using Runner = facebook::velox::exec::test::AggregationFuzzerRunner;
 
   Runner::Options options;
   options.onlyFunctions = FLAGS_only;
   options.skipFunctions = skipFunctions;
   options.customVerificationFunctions = customVerificationFunctions;
+  options.customInputGenerators = customInputGenerators;
```
Comment (pedroerp) [inline velox/exec/tests/AggregationFuzzerTest.cpp:247]: nit: could also move to a helper function for clarity:
```
options.customInputGenerators = getCustomInputGenerators();
```

File: velox/exec/tests/utils/AggregationFuzzer.cpp [hunk 5]
```diff
@@ -185,29 +195,39 @@ class AggregationFuzzer {
...
   // row. Row numbers start from 0. This additional input vector is needed for
   // result verification of window aggregations.
   std::vector<RowVectorPtr> generateInputDataWithRowNumber(
       std::vector<std::string> names,
-      std::vector<TypePtr> types);
+      std::vector<TypePtr> types,
+      const CallableSignature& signature);
 
-  void verifyWindow(
+  bool verifyWindow(
```
Comment (pedroerp) [inline velox/exec/tests/utils/AggregationFuzzer.cpp:222]: maybe document what the return value means.

**Review by pedroerp — APPROVED** [review]
A few nits and small comments, but looks solid. Thanks!

**Comment by mbasmanova** [comment]
@pedroerp Pedro, than you for review and nice suggestions. Applied and updated.

**Pull request closed without merging**